truffle-config.js

```diff
@@ -42,61 +46,95 @@ module.exports = {
-      network_id: "3"
-    }
+
+
     // Another network with more advanced options...
     // advanced: {
-    // port: 8777,             // Custom port
-    // network_id: 1342,       // Custom network
-    // gas: 8500000,           // Gas sent with each transaction (default: ~6700000)
-    // gasPrice: 20000000000,  // 20 gwei (in wei) (default: 100 gwei)
-    // from: <address>,        // Account to send txs from (default: accounts[0])
-    // websockets: true        // Enable EventEmitter interface for web3 (default: false)
+      // port: 8777,             // Custom port
+      // network_id: 1342,       // Custom network
+      // gas: 8500000,           // Gas sent with each transaction (default: ~6700000)
+      // gasPrice: 20000000000,  // 20 gwei (in wei) (default: 100 gwei)
+      // from: <address>,        // Account to send txs from (default: accounts[0])
+      // websockets: true        // Enable EventEmitter interface for web3 (default: false)
     // },
+
     // Useful for deploying to a public network.
     // NB: It's important to wrap the provider as a function.
-    // ropsten: {
-    // provider: () => new HDWalletProvider(mnemonic, `https://ropsten.infura.io/v3/YOUR-PROJECT-ID`),
-    // network_id: 3,       // Ropsten's id
-    // gas: 5500000,        // Ropsten has a lower block limit than mainnet
-    // confirmations: 2,    // # of confs to wait between deployments. (default: 0)
-    // timeoutBlocks: 200,  // # of blocks before a deployment times out  (minimum/default: 50)
-    // skipDryRun: true     // Skip dry run before migrations? (default: false for public nets )
-    // },
+
+    kovan: {
+      provider: () => getProvider("kovan"),
+      network_id: 42,         // Kovan's id
+      confirmations: 2,       // # of confs to wait between deployments. (default: 0)
+      timeoutBlocks: 200,     // # of blocks before a deployment times out  (minimum/default: 50)
+      gasPrice: 10000000000,  // 10 gwei
+      skipDryRun: false       // Skip dry run before migrations? (default: false for public nets )
+    },
+
+    goerli: {
+      provider: () => getProvider("goerli"),
+      network_id: 5,          // Goerli's id
+      confirmations: 2,       // # of confs to wait between deployments. (default: 0)
+      timeoutBlocks: 200,     // # of blocks before a deployment times out  (minimum/default: 50)
+      gasPrice: 10000000000,  // 10 gwei
+      skipDryRun: false       // Skip dry run before migrations? (default: false for public nets )
+    },
+
+    rinkeby: {
+      provider: () => getProvider("rinkeby"),
+      network_id: 4,          // Rinkeby's id
+      confirmations: 0,       // # of confs to wait between deployments. (default: 0)
+      timeoutBlocks: 200,     // # of blocks before a deployment times out  (minimum/default: 50)
+      gasPrice: 10000000000,  // 10 gwei
+      skipDryRun: false       // Skip dry run before migrations? (default: false for public nets )
+    },
+
+    ropsten: {
```

Comment:
Not a blocker, but Ropsten has been having a lot of [issues](https://github.com/openethereum/openethereum/issues/11862) lately, leading to some major platforms (including Chainlink) moving to Kovan for testnet stuff. Steer clear of deploying to Ropsten for now.